go/teams/conflicts.go

```diff
@@ -0,0 +1,115 @@
+package teams
+
+import (
+	"fmt"
+	"github.com/keybase/client/go/libkb"
+	"github.com/keybase/client/go/lru"
+	"github.com/keybase/client/go/protocol/keybase1"
+	"golang.org/x/net/context"
+)
+
+// Increment to invalidate the disk cache.
+const diskStorageVersionConflictInfo = 1
+
+type conflictID struct {
+	isPublic bool
+	id       keybase1.TeamID
+}
+
+func (i conflictID) MemKey() string {
+	prefix := "r"
+	if i.isPublic {
+		prefix = "u"
+	}
+	return fmt.Sprintf("%s%s", prefix, i.id)
+}
+
+func (i conflictID) DbKey() libkb.DbKey {
+	return libkb.DbKey{
+		Typ: libkb.ObjType(libkb.DBImplicitTeamConflictInfo),
+		Key: i.MemKey(),
+	}
+}
+
+var _ libkb.LRUKeyer = conflictID{}
+
+type rawGetConflictInfo struct {
+	Status       libkb.AppStatus                    `json:"status"`
+	ConflictInfo *keybase1.ImplicitTeamConflictInfo `json:"conflict_info"`
+}
+
+func (r *rawGetConflictInfo) GetAppStatus() *libkb.AppStatus {
+	return &r.Status
+}
+
+func GetConflictInfo(ctx context.Context, g *libkb.GlobalContext, id keybase1.TeamID, isFullyResolved bool, name keybase1.ImplicitTeamDisplayName) (ret keybase1.ImplicitTeamDisplayName, err error) {
+	defer g.CTrace(ctx, fmt.Sprintf("GetConflictInfo(%s,%v)", id, name), func() error { return err })()
+
+	ret = name.DeepCopy()
+
+	key := conflictID{name.IsPublic, id}
+	cv, err := g.GetImplicitTeamConflictInfoCacher().Get(ctx, g, key)
+	if err != nil {
+		g.Log.CWarningf(ctx, "In fetching from cache: %s", err.Error())
+	}
+	if cv != nil {
+		if p, ok := cv.(*keybase1.ImplicitTeamConflictInfo); ok {
+			if p.IsConflict() {
+				ret.ConflictInfo = p
+			}
+			return ret, nil
+		}
+		g.Log.CWarningf(ctx, "Bad element of wrong type from cache: %T", cv)
```

Comment:
That `cv` is an interface{} out here is unfortunate. Wrap LRUer in a type for conflict infos? Then `GetImplicitTeamConflictInfoCacher` would return that type

Reply:
I don't think that's really necessary.  It's the same thing, we're going to need this cast somewhere.